source/rules/rxjsSuffixSubjectsRule.ts

```diff
@@ -0,0 +1,129 @@
+/**
+ * @license Use of this source code is governed by an MIT-style license that
+ * can be found in the LICENSE file at https://github.com/cartant/rxjs-tslint-rules
+ */
+/* tslint:disable:no-use-before-declare */
+
+import { tsquery } from "@phenomnomnominal/tsquery";
+import * as Lint from "tslint";
+import * as ts from "typescript";
+import { couldBeType } from "../support/util";
+
+const defaultTypesRegExp = /^EventEmitter$/;
+
+export class Rule extends Lint.Rules.TypedRule {
+  public static metadata: Lint.IRuleMetadata = {
+    description:
+      "Ensures subjects are suffixed with suffix specified by `suffix` option.",
+    options: {
+      properties: {
+        functions: { type: "boolean" },
+        methods: { type: "boolean" },
+        parameters: { type: "boolean" },
+        properties: { type: "boolean" },
+        suffix: { type: "string" },
+        types: { type: "object" },
+        variables: { type: "boolean" }
+      },
+      type: "object"
+    },
+    optionsDescription: Lint.Utils.dedent`
+      An optional object with optional \`parameters\`, \`properties\` and \`variables\` properties.
+      The properies are booleans and determine whether or not subjects of that particular expression need a suffix.
+      \`parameters\`, \`properties\` and \`variables\` default to \`true\`, and the default suffix is 'Subject'.
+      The object also has optional \`types\` properties which are themselves
+      objects containing keys that are regular expressions and values that are booleans -
+      indicating whether suffixing is required for particular types.`,
+    requiresTypeInfo: true,
+    ruleName: '"rxjs-suffix-subjects"',
+    type: "style",
+    typescriptOnly: true
+  };
+
+  private suffix = 'Subject';
+  private suffixRegex = new RegExp(`${this.suffix}\\$?$`, "i");
+  private types: { regExp: RegExp; validate: boolean }[] = [];
+  private validate = {
+    parameters: true,
+    properties: true,
+    variables: true
+  };
+  public readonly FAILURE_MESSAGE = (identifier: string) =>
```

Comment:
Given that the rule is no longer using a walker, these private properties don't really need to be class properties. They could be moved into the `applyWithProgram` function. That would be simpler.